app/src/main/java/com/es/findsoccerplayers/LoginActivity.java

```diff
@@ -115,79 +128,66 @@ public void onClick(View v) {
         });
     }
 
+    @Override
+    protected void onStart() {
+        super.onStart();
+        Log.w("LoginAct","Act Creata");
+    }
+
+    private void signIn() {
+        prgBar.setVisibility(View.VISIBLE);
+        Intent signInIntent = mGoogleSignInClient.getSignInIntent();
+        //Chiama la funzione onActivityResult
+        startActivityForResult(signInIntent, RC_SIGN_IN);
+    }
+
+    //Funzione che gestisce il tentativo di login
     @Override
     public void onActivityResult(int requestCode, int resultCode, Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
-        // Result returned from launching the Intent from GoogleSignInApi.getSignInIntent(...);
+        //Se il requestCode è lo stesso di quello passato dalla funzione signIn()
         if (requestCode == RC_SIGN_IN) {
             Task<GoogleSignInAccount> task = GoogleSignIn.getSignedInAccountFromIntent(data);
             try {
-                // Google Sign In was successful, authenticate with Firebase
                 GoogleSignInAccount account = task.getResult(ApiException.class);
                 firebaseAuthWithGoogle(account.getIdToken());
                 Toast.makeText(LoginActivity.this, "Loggato con successo", Toast.LENGTH_SHORT).show();
```

Comment:
copia dentro qua il codice di `firebaseAuthWithGoogle`, tanto sono poche righe. E io farei anche il `finish()` dell'activity di login, tanto è improbabile che l'utente appena loggato faccia il log out e torni all'activity di login.
In quel caso ci sta ricreare l'activity